catalog/nsili/catalog-nsili-source/src/main/java/org/codice/alliance/nsili/source/NsiliSource.java

```diff
@@ -0,0 +1,921 @@
+            LOGGER.warn("{} : Received empty attributes list from STANAG source.", getId());
+        }
+
+        this.sortableAttributes = sortableAttributesMap;
+        this.resultAttributes = resultAttributesMap;
+    }
+
+    /**
+     * Obtains all queryable attributes for all possible views that the Federated Source can provide.
+     * According to ANNEX D, TABLE D-6, the passed parameter in get_view_names is an empty list(not used).
+     *
+     * @return a map of each view and the queryable attributes provided by the source for that view
+     */
+    private void initQueryableAttributes() {
+        if (views == null || views.length == 0) {
+            return;
+        }
+        HashMap<String, List<AttributeInformation>> map = new HashMap<>();
+
+        try {
+            for (int i = 0; i < views.length; i++) {
+                AttributeInformation[] attributeInformationArray =
+                        dataModelMgr.get_queryable_attributes(views[i].view_name, new NameValue[0]);
+                List<AttributeInformation> attributeInformationList = new ArrayList<>();
+                for (int c = 0; c < attributeInformationArray.length; c++) {
+                    attributeInformationList.add(attributeInformationArray[c]);
+                }
+                map.put(views[i].view_name, attributeInformationList);
+            }
+        } catch (ProcessingFault | SystemFault | InvalidInputParameter e) {
+            LOGGER.error("{} : Unable to retrieve queryable attributes.", id, e);
+        }
+
+        if (map.size() == 0) {
+            LOGGER.warn("{} : Received empty queryable attributes from STANAG source.", getId());
+        }
+        queryableAttributes = map;
+    }
+
+    /**
+     * Obtains the description of the source from the Library interface.
+     *
+     * @return a description of the source
+     */
+    private void setSourceDescription() {
+        StringBuilder stringBuilder = new StringBuilder();
+        try {
+            LibraryDescription libraryDescription = library.get_library_description();
+            stringBuilder.append(libraryDescription.library_name + " : ");
+            stringBuilder.append(libraryDescription.library_description);
+        } catch (ProcessingFault | SystemFault e) {
+            LOGGER.error("{} : Unable to retrieve source description.", id, e);
+        }
+        String description = stringBuilder.toString();
+        if (StringUtils.isBlank(description)) {
+            LOGGER.warn("{} :  Unable to retrieve source description.", getId());
+        }
+        this.description = description;
+    }
+
+    public void destroy() {
+        orb.shutdown(true);
+        availabilityPollFuture.cancel(true);
+        scheduler.shutdownNow();
+    }
+
+    public void refresh(Map<String, Object> configuration) {
+        LOGGER.debug("Entering Refresh : {}", getId());
+
+        if (configuration == null || configuration.isEmpty()) {
```

Comment:
you could use MapUtils.isEmpty() here.  https://commons.apache.org/proper/commons-collections/apidocs/org/apache/commons/collections4/MapUtils.html#isEmpty(java.util.Map)